examples/chip-tool-darwin/templates/commands.zapt

```diff
@@ -0,0 +1,282 @@
+
+        {{#chip_cluster_command_arguments}}
+        {{#first}}
+        __auto_type * params = [[CHIP{{asUpperCamelCase parent.clusterName}}Cluster{{asUpperCamelCase parent.name}}Params alloc] init];
+        {{/first}}
+
+        {{~#*inline "field"}}m{{asUpperCamelCase label}}{{/inline~}}
+        {{#if isArray}}
+        // {{label}} Array parsing is not supported yet
+        {{else if isStruct}}
+        // {{label}} Struct parsing is not supported yet
+        {{else if (isOctetString type)}}
+        params.{{asStructPropertyName label}} = [[NSData alloc] initWithBytes:{{>field}}.data() length:{{>field}}.size()];
+        {{else if (isString type)}}
+        params.{{asStructPropertyName label}} = [[NSString alloc] initWithBytes:{{>field}}.data() length:{{>field}}.size() encoding:NSUTF8StringEncoding];
+        {{else}}
+        params.{{asStructPropertyName label}} = [NSNumber numberWith{{asObjectiveCNumberType "" type false}}:{{>field}}];
+        {{/if}}
+        {{/chip_cluster_command_arguments}}
+        [cluster {{asLowerCamelCase name}}With{{#if (hasArguments)}}Params:params completionHandler:{{else}}CompletionHandler:{{/if}}
+        {{#if hasSpecificResponse}}
+            ^(CHIP{{asUpperCamelCase clusterName}}Cluster{{asUpperCamelCase responseName}}Params * _Nullable values, NSError * _Nullable error) {
+                 NSLog(@"Values: %@", values);
+        {{else}}
+            ^(NSError * _Nullable error) {
+        {{/if}}
+            SetCommandExitStatus([CHIPError errorToCHIPErrorCode:error]);
+            }];
+
+        return CHIP_NO_ERROR;
+    }
+
+private:
+    {{#chip_cluster_command_arguments}}
+    {{#if isArray}}
+    {{else if isStruct}}
+    {{else if (isOctetString type)}}
+    chip::ByteSpan m{{asUpperCamelCase label}};
+    {{else if (isCharString type)}}
+    chip::ByteSpan m{{asUpperCamelCase label}};
+    {{else}}
+    {{chipType}} m{{asUpperCamelCase label}};
+    {{/if}}
+    {{/chip_cluster_command_arguments}}
+};
+
+{{/chip_cluster_commands}}
+
+{{#chip_server_cluster_attributes}}
+{{#*inline "attribute"}}Attribute{{asUpperCamelCase name}}{{/inline}}
+
+/*
+ * Attribute {{asUpperCamelCase name}}
+ */
+class Read{{asUpperCamelCase parent.name}}{{asUpperCamelCase name}}: public ModelCommand
+{
+public:
+    Read{{asUpperCamelCase parent.name}}{{asUpperCamelCase name}}(): ModelCommand("read")
+    {
+        AddArgument("attr-name", "{{asDelimitedCommand (asUpperCamelCase name)}}");
+        ModelCommand::AddArguments();
+    }
+
+    ~Read{{asUpperCamelCase parent.name}}{{asUpperCamelCase name}}()
+    {
+    }
+
+    CHIP_ERROR SendCommand(CHIPDevice * device, chip::EndpointId endpointId) override
+    {
+        ChipLogProgress(chipTool, "Sending cluster ({{asHex parent.code 8}}) ReadAttribute ({{asHex code 8}}) on endpoint %" PRIu16, endpointId);
+
+        {{! TODO: Various types (floats, structs) not supported here. }}
```

Comment:
Floats don't have a callback name of "Unsupported".  The only thing that does is structs.

Reply:
Updated. Thank you.